src/entities/mellowLpRouter.ts

```diff
@@ -453,6 +453,157 @@ class MellowLpRouter {
+    if (
+      isUndefined(this.readOnlyContracts) ||
+      isUndefined(this.writeContracts) ||
+      isUndefined(this.userAddress)
+    ) {
+      throw new Error('Uninitialized contracts.');
+    }
+
+    const subvaultsCount: number = (
+      await this.readOnlyContracts.erc20RootVault[vaultIndex].subvaultNfts()
+    ).length;
+
+    const minTokenAmounts = BigNumber.from(0);
+    const vaultsOptions = new Array(subvaultsCount).fill(0x0);
+
+    console.log(`Calling claimLPTokens(${vaultIndex}, ${[minTokenAmounts]}, [${vaultsOptions}])`);
+
+    try {
+      await this.writeContracts.mellowRouter.callStatic.claimLPTokens(
+        vaultIndex,
+        [minTokenAmounts],
+        vaultsOptions,
+      );
+    } catch (err) {
+      console.error('Error during claimLPTokens:', err);
+      sentryTracker.captureException(err);
+      sentryTracker.captureMessage('Unsuccessful claimLPTokens simulation.');
+      throw new Error('Unsuccessful claimLPTokens simulation.');
+    }
+
+    const gasLimit = await this.writeContracts.mellowRouter.estimateGas.claimLPTokens(
+      vaultIndex,
+      [minTokenAmounts],
+      vaultsOptions,
+    );
+
+    const tx = await this.writeContracts.mellowRouter.claimLPTokens(
+      vaultIndex,
+      [minTokenAmounts],
+      vaultsOptions,
+      {
+        gasLimit: getGasBuffer(gasLimit),
+      },
+    );
+
+    try {
+      const receipt = await tx.wait();
+
+      try {
+        await this.refreshWalletBalance();
+      } catch (err) {
+        sentryTracker.captureException(err);
+        sentryTracker.captureMessage('Wallet user balance failed to refresh after withdrawal');
+        console.error('Wallet user balance failed to refresh after withdraw');
+      }
+
+      try {
+        await this.refreshUserComittedDeposit();
+      } catch (err) {
+        sentryTracker.captureException(err);
+        sentryTracker.captureMessage('User deposit failed to refresh after withdrawal');
+        console.error('User deposit failed to refresh after withdraw');
+      }
+
+      return receipt;
+    } catch (err) {
+      sentryTracker.captureException(err);
+      sentryTracker.captureMessage('Unsucessful withdrawal confirmation.');
+      throw new Error('Unsucessful withdraw confirmation.');
+    }
+  };
+
+  rollover = async (vaultIndex: number): Promise<ContractReceipt> => {
+    if (
+      isUndefined(this.readOnlyContracts) ||
+      isUndefined(this.writeContracts) ||
+      isUndefined(this.userAddress)
+    ) {
+      throw new Error('Uninitialized contracts.');
+    }
+
+    const subvaultsCount: number = (
+      await this.readOnlyContracts.erc20RootVault[vaultIndex].subvaultNfts()
+    ).length;
+
+    const minTokenAmounts = BigNumber.from(0);
+    const vaultsOptions = new Array(subvaultsCount).fill(0x0);
+
+    console.log(
+      `Calling rolloverLPTokens(${vaultIndex}, ${[minTokenAmounts]}, [${vaultsOptions}], ${
+        this.defaultWeights
```

Comment:
So we are going to rollover with the same weights used for any other new deposit, right?